apps/dashboard/app/javascript/packs/files/DataTable.js

```diff
@@ -0,0 +1,431 @@
+import 'datatables.net';
+import 'datatables.net-bs4/js/dataTables.bootstrap4';
+import 'datatables.net-select';
+import 'datatables.net-select-bs4';
+import Handlebars from 'handlebars';
+import {} from './SweetAlert.js';
+import {} from './FileOps.js';
+import {} from './UppyOps.js';
+import {} from './ClipBoard.js';
+
+
+let table = null;
+
+$(document).ready(function () {
+    table = new DataTable();
+
+    /* BUTTON ACTIONS */
+    $("#new-file-btn").on("click", function () {
+        $("#directory-contents").trigger('fileOpsNewFile');
+    });
+
+    $("#new-folder-btn").on("click", function () {
+        $("#directory-contents").trigger('fileOpsNewFolder');
+    });
+
+    $("#download-btn").on("click", function () {
+        let selection = table.getTable().rows({ selected: true }).data();
+        if(selection.length == 0) {
+            const eventData = {
+                'title': 'Select a file, files, or directory to download',
+                'message': 'You have selected none.',
+            };
+
+            $("#directory-contents").trigger('swalShowError', eventData);
+
+        } else {
```

Comment:
Why not push this knowledge to the listener? Let the listener deal with the event the way _they_ want to. That way this file won't have to worry about swal errors.

Reply:
> Why not push this knowledge to the listener? Let the listener deal with the event the way _they_ want to. That way this file won't have to worry about swal errors.

done.